### PR TITLE
Support bulk read for abstractQueuedStreams

### PR DESCRIPTION
**Pull request by no2chem**
This patch moves the implementation of remainingUpTo into the AbstractContextStreamView,
allowing subclasses to implement bulk reads by implementing getNextEntries. 

AbstractContextStreamView continues to manage copy-on-write contexts.

This paves the way for bulk reads to be supported by the address view, and eventually,
the log-unit/log unit client.

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java [hunk 1]
```diff
@@ -110,18 +110,74 @@ final public synchronized ILogData nextUpTo(final long maxGlobal) {
...
+            // sure we get up to maxGlobal.
+            entries.addAll(remainingUpTo(maxGlobal));
+
+            // Update the pointer with the last entry
+            updatePointer(entries.get(entries.size() - 1));
+
+            return entries;
+        }
+
+        // Otherwise update the pointer with the last entry
```
Comment (dahliamalkhi) [inline runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java]: make this the 'if' case, and return. Then if you didn't return, handle nested contexts.

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **97784ca**.

**Throughput**: _Average_: 34.4827586207, _Standard Deviation_: 10.5458626947
**Latency**: _Average_: 279.295, _Standard Deviation_: 118.185007404

 An interactive dashboard with Pull Request Performance Metrics is available at:
[Pull Request #446 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/nxjGbyDXqbSakmmJZXInwckiCwlFaLru)

File: runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java (hunk 1 above)
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java]: done

**Comment by no2chem** [comment]
Agree about the stream contexts. It's only tested for by a single unit test.

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **d1e9dcb**.

**Throughput**: _Average_: 37.037037037 [executions/sec], _Standard Deviation_: 10.4933187967
**Latency**: _Average_: 263.879 [ms], _Standard Deviation_: 110.332462852

 An interactive dashboard with Pull Request Performance Metrics is available at:
[Pull Request #446 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/FIcCOvC62pG49zUu2fa3Bsszi4xIXP1U)

**Pull request merged**